stdlib_test.go

```diff
@@ -48,214 +49,271 @@ func TestStdlibAdapterUsage(t *testing.T) {
 	}
 }
 
-func TestStdLibAdapterExtraction(t *testing.T) {
-	buf := &bytes.Buffer{}
-	logger := NewLogfmtLogger(buf)
-	writer := NewStdlibAdapter(logger)
-	for input, want := range map[string]string{
-		"hello":                             "msg=hello\n",
-		"2009/01/23: hello":                 "ts=2009/01/23 msg=hello\n",
-		"2009/01/23 01:23:23: hello":        "ts=\"2009/01/23 01:23:23\" msg=hello\n",
-		"01:23:23: hello":                   "ts=01:23:23 msg=hello\n",
-		"2009/01/23 01:23:23.123123: hello": "ts=\"2009/01/23 01:23:23.123123\" msg=hello\n",
-		"2009/01/23 01:23:23.123123 /a/b/c/d.go:23: hello": "ts=\"2009/01/23 01:23:23.123123\" caller=/a/b/c/d.go:23 msg=hello\n",
-		"01:23:23.123123 /a/b/c/d.go:23: hello":            "ts=01:23:23.123123 caller=/a/b/c/d.go:23 msg=hello\n",
-		"2009/01/23 01:23:23 /a/b/c/d.go:23: hello":        "ts=\"2009/01/23 01:23:23\" caller=/a/b/c/d.go:23 msg=hello\n",
-		"2009/01/23 /a/b/c/d.go:23: hello":                 "ts=2009/01/23 caller=/a/b/c/d.go:23 msg=hello\n",
-		"/a/b/c/d.go:23: hello":                            "caller=/a/b/c/d.go:23 msg=hello\n",
-	} {
-		buf.Reset()
-		fmt.Fprint(writer, input)
-		if have := buf.String(); want != have {
-			t.Errorf("%q: want %#v, have %#v", input, want, have)
-		}
+func TestStdLibAdapter(t *testing.T) {
+	var testcases = []struct {
+		regexp                *regexp.Regexp
+		input                 string
+		want                  string
+		prefix                string
+		prefixJoinPrefixToMsg bool
```

Comment:
```suggestion
		input      string
		regexp     regexp.Regexp
		prefix     string
		joinPrefix bool
		want       string
```

Reply:
You'll also need to update the test cases.

Reply:
Yeah, I do it rn